explorer/ast/statement.cpp

```diff
@@ -15,111 +15,136 @@ using llvm::cast;
 
 Statement::~Statement() = default;
 
-void Statement::PrintDepth(int depth, llvm::raw_ostream& out) const {
+void Statement::PrintDepth(int depth, int indent_num_spaces,
+                           llvm::raw_ostream& out) const {
   if (depth == 0) {
-    out << " ... ";
+    out.indent(indent_num_spaces) << " ... ";
```

Comment:
It looks like every form of statement starts with an indent, except for Block.

Second, what do you think of calling `out.indent` once at the start of this PrintDepth function, instead of at the start of each statement? Given `Block`, something like:

```
if (kind() != Statement::Block) {
  out.indent(...);
}
```

Note though, we'll probably support actual block statements down the line (doesn't seem to parse right now, surprisingly), for scoping reasons. e.g.:

```
fn F() {
  // Block statement:
  {
     var scoped: MyType = ...;
  }
  // Code that depends on `scoped` being destroyed goes here.
}
```

And then those should be indented too.

Reply:
Note on indenting blocks, in some cases like `while` it's going to be after a newline, won't it? i.e., you might want to test, but I think you'd get:

```
fn F() {
  while (...)
{
    // while body is indented, open curly is not.
  }
}
```

Reply:
Oh yes, these issues can arise with the current way of indentation. Thanks for pointing this out!
I was thinking can we have formatting like this?

```
fn Foo() -> type
{
  // body
  if(...)
  {
    ...
  }
}
```

instead of 
```
fn Foo() -> type {
  // body
  if(...) {
    ...
  }
}
```

It would be a bit simpler to implement.
What do you think?

Reply:
I think that'd be fine, yes.